File: applications/wordcount/wc_mapper.py
import sys
from pmr.mapper import Mapper
    
if __name__ == "__main__":
    # Initialize Map Job
    mapJob = Mapper(sys.argv)
          
    
    # map function    
    with open(mapJob.chunkFile) as fh:
        line = fh.read()
        for word in line.split():
            mapJob.emit(word, "%s,%s" % (word, 1))
                            
    # # Finalize map job  
    mapJob.finalize()

File: applications/java/wordcount/WcMapper.java
package pmr;

import java.io.BufferedReader;
import java.io.FileReader;

import pmr.Mapper;

public class WcMapper {
	public static void main(String[] args) throws Exception {

		// Initialize Map Job
		Mapper mapJob = new Mapper(args);

		// Map function
		BufferedReader br = new BufferedReader(new FileReader(mapJob.chunkFile));
		String line;

		while ((line = br.readLine()) != null) {
			for (String word : line.split(" ")) {
				mapJob.emit(word, 1);
			}
		}
		br.close();

		// Finalize map job
		mapJob.finalize();

	}
}
